apps/documents/views.py

```diff
@@ -0,0 +1,231 @@
+        search = self.request.GET.get("search")
+        if search:
+            # TODO: Expand to search summary as well
+            name_similarity = TrigramSimilarity("name", search)
+            queryset = (
+                queryset.annotate(similarity=name_similarity).filter(Q(similarity__gt=0.2)).order_by("-similarity")
+            )
+        return queryset
+
+
+class FileDetails(LoginAndTeamRequiredMixin, BaseDetailsView):
+    template_name = "documents/file_details.html"
+    model = File
+    permission_required = "files.view_file"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        file = self.get_object()
+        collection_names = file.repository_set.filter(type=RepositoryType.COLLECTION).values_list("name", flat=True)
+        context["current_collections"] = list(collection_names)
+        context["edit_url"] = reverse(
+            "documents:edit_file", kwargs={"team_slug": self.kwargs["team_slug"], "id": file.id}
+        )
+        context["delete_url"] = reverse(
+            "documents:delete_file", kwargs={"team_slug": self.kwargs["team_slug"], "id": file.id}
+        )
+        context["available_collections"] = Repository.objects.filter(
+            team__slug=self.kwargs["team_slug"], type=RepositoryType.COLLECTION
+        ).values_list("name", flat=True)
+        return context
+
+
+@require_POST
+@transaction.atomic()
+@permission_required("files.add_file")
+def upload_files(request, team_slug: str):
+    """Upload files to a collection"""
+    # TODO: Check collection size and error if it's too large with the new files added
+    files = []
+    file_summaries = json.loads(request.POST["file_summaries"])
+    for uploaded_file in request.FILES.getlist("files"):
+        files.append(
+            File.objects.create(
+                team=request.team,
+                name=uploaded_file.name,
+                file=uploaded_file,
+                summary=file_summaries[uploaded_file.name],
+            )
+        )
+    repo, _ = Repository.objects.get_or_create(
+        type=RepositoryType.COLLECTION, team=request.team, name=request.POST.get("collection_name")
+    )
+    repo.files.add(*files)
+    return redirect(reverse("documents:repositories", kwargs={"team_slug": team_slug, "tab_name": "files"}))
+
+
+@login_and_team_required
+@permission_required("files.delete_file")
+def delete_file(request, team_slug: str, id: int):
+    file = get_object_or_404(File, team__slug=team_slug, id=id)
+    file.delete()
+    return redirect(reverse("documents:repositories", kwargs={"team_slug": team_slug, "tab_name": "files"}))
+
+
+@login_and_team_required
+@require_POST
+@permission_required("files.change_file")
+def edit_file(request, team_slug: str, id: int):
+    file = get_object_or_404(File.objects.defer("file"), team__slug=team_slug, id=id)
```

Comment:
I don't think the 'defer' is necessary here. It won't actually read the file when you load it from the DB. It will just contain the path / reference to the file.